src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1201,7 +1201,7 @@ public static bool IsParameterTypeContext(this SyntaxTree syntaxTree, int positi
 
             if (token.IsKind(SyntaxKind.OutKeyword) &&
                 token.Parent.IsKind(SyntaxKind.Argument) &&
-                ((ArgumentSyntax)token.Parent).Declaration != null)
```

Comment:
So, i'm ok with this.  However, i did want to point out that we have a different philosophy on the IDE side about this wrt to suggestive features like IntelliSense.  In those cases we prefer the minimal check that expresses the scenario, with the knowledge that maybe in the future that might mean the feature lights up in more scenarios.  We don't try to over-constrain things.

I feel like you approach is eminently suited for the compiler space, whereas there are different things to weigh off when doing IDE features :)

Anyways. :+1:   Just keep this in mind in the future.  Thanks!